.github/workflows/ci.yml

```diff
@@ -42,12 +42,16 @@ jobs:
       timeout-minutes: 5
       run: |
         set -x
-        RESULT=" "
-        while [[ "$RESULT" != "  HTTP/1.1 200 OK" ]]; do
-          sleep 1
-          RESULT=$(kubectl exec deployments/frontend -- sh -c "wget --spider -S "http://frontend" 2>&1 | grep 'HTTP/'")
-          echo "front end response: $RESULT"
+        # start fresh loadgenerator pod
+        kubectl delete pod -l app=loadgenerator
+        # wait for requests to come in
+        REQUEST_COUNT="0"
+        while [[ "$REQUEST_COUNT"  -lt "50"  ]]; do
+            sleep 5
+            REQUEST_COUNT=$(kubectl logs -l app=loadgenerator | grep Aggregated | awk '{print $2}')
         done
-        if [[ "$RESULT" != "  HTTP/1.1 200 OK" ]]; then 
+        # ensure there are no errors hitting endpoints
+        ERROR_COUNT=$(kubectl logs -l app=loadgenerator | grep Aggregated | awk '{print $3}' | sed "s/[(][^)]*[)]//g")
```

Comment:
a comment about what the sed string does could be helpful (-:

Reply:
yeah, similarly I'm not sure which part of the command actually ensures this matches only 1 occurrence of `Aggregated`.

Reply:
For example my `kubectl logs -l app=loadgenerator -c main` on cluster `demo-app` on project `microservices-demo-app` right now returns:



```
$ kubectl logs -l app=loadgenerator -c main
 GET /product/66VCHSJNUP                                       119217    39(0.03%)      88      25    2735  |      40    0.30
 GET /product/6E92ZMYYFZ                                       119973    37(0.03%)      87      27    3360  |      40    0.20
 GET /product/9SIQT8TOJO                                       119565    33(0.03%)      87      27    3230  |      40    0.20
 GET /product/L9ECAV7KIM                                       119289    24(0.02%)      88      27    2828  |      39    0.10
 GET /product/LS4PSXUNUM                                       119623    36(0.03%)      88      26    2995  |      39    0.00
 GET /product/OLJCESPC7Z                                       119846    44(0.04%)      89      26    2970  |      40    0.00
 POST /setCurrency                                             165396    47(0.03%)     114      36    3094  |      69    0.20
--------------------------------------------------------------------------------------------------------------------------------------------
 Total                                                        1902768   558(0.03%)                                       1.60
```

So this command above returns 0, despite 558 failures.

If I actually tail with `-f` I get more lines.

Even when loadgenerator prints its output every few secs once I do `kubectl logs -f`, I still don't see word `Aggregated`. I assume it's printed at the beginning, then lost.

Reply:
@ahmetb it looks like you're using an older version of locust. It was updated to 0.13 [here](https://github.com/GoogleCloudPlatform/microservices-demo/commit/997b5e88072d0948a93b40ccf8c27eb1fcb66ebe#diff-dd74128e3994eeda8fd49072b5901c42), which uses [Aggregated instead of Total](https://docs.locust.io/en/stable/changelog.html#id8)

Reply:
ah yeah, thanks. I guess I forgot we don't auto-deploy the new versions to the demo cluster.